aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp

```diff
@@ -812,31 +900,67 @@ void svd_cusolver(const Tensor& A,
   const auto n = A.size(-1);
   const auto k = std::min(m, n);
 
-  // heuristic for using `gesvdjBatched` over `gesvdj`
-  // gesvdjBatched just works for square matrices
-  // Need to pass a copy of A, since A will be rewritten inside the function call
-  if (m <= 32 && n <= 32 && batch_size > 1 && (full_matrices || m == n)) {
-    svd_cusolver_gesvdjBatched(cloneBatchedColumnMajor(A), U, S, V, info, compute_uv);
-  } else {
-    svd_cusolver_gesvdj(cloneBatchedColumnMajor(A), U, S, V, info, full_matrices, compute_uv);
-  }
+  static const char* check_svd_doc = "Check doc at https://pytorch.org/docs/stable/generated/torch.linalg.svd.html";
 
-  // A device-host sync will be performed.
-  const auto gesvdj_convergence_check = _check_gesvdj_convergence(info, k + 1);
+  const bool gesvdj_batched_requirements = (m <= 32 && n <= 32 && batch_size > 1 && (full_matrices || m == n));
+  bool convergence_check_needed = false;
 
-  if (!gesvdj_convergence_check.empty()) {
-    // fall back to gesvd path for those non-converging batches
+  if (!driver.has_value()) {
+    // use the default heuristics
+    if (gesvdj_batched_requirements) {
+      svd_cusolver_gesvdjBatched(cloneBatchedColumnMajor(A), U, S, V, info, compute_uv);
+      convergence_check_needed = true;
+    } else {
+      // gesvdj driver may be numerically unstable for large sized matrix
+      svd_cusolver_gesvdj(cloneBatchedColumnMajor(A), U, S, V, info, full_matrices, compute_uv);
+      convergence_check_needed = true;
+    }
```

Comment:
The default heuristics are simply "driver == "gesvdj". It'd be simpler to do
```c++
// The default heuristic is to use gesvdj
auto driver_v = driver.value_or("gesvdj");
```
and get rid of this branch.